mountpoint-s3-client/tests/auth.rs

```diff
@@ -240,6 +240,130 @@ async fn test_profile_provider_assume_role_async() {
+
+    // Create two source profiles to provide credentials from previously created files using `credential_process`.
+    let (correct_source_profile, incorrect_source_profile) = {
+        let correct = "correct-source-profile";
+        writeln!(config_file, "[profile {}]", correct).unwrap();
+        writeln!(
+            config_file,
+            "credential_process=cat {}",
+            correct_credential_file.path().to_string_lossy()
+        )
+        .unwrap();
+        let incorrect = "incorrect-source-profile";
+        writeln!(config_file, "[profile {}]", incorrect).unwrap();
+        writeln!(
+            config_file,
+            "credential_process=cat {}",
+            incorrect_credential_file.path().to_string_lossy()
+        )
+        .unwrap();
+        (correct, incorrect)
+    };
+
+    // Create two profiles to assume our test role with previously created source profiles.
+    let (correct_profile, incorrect_profile) = {
+        let correct = "correct-profile";
+        writeln!(config_file, "[profile {}]", correct).unwrap();
+        writeln!(config_file, "role_arn={}", get_subsession_iam_role()).unwrap();
+        writeln!(config_file, "source_profile={}", correct_source_profile).unwrap();
+        writeln!(config_file, "region={}", &get_test_region()).unwrap();
+        let incorrect = "incorrect-profile";
+        writeln!(config_file, "[profile {}]", incorrect).unwrap();
+        writeln!(config_file, "role_arn={}", get_subsession_iam_role()).unwrap();
+        writeln!(config_file, "source_profile={}", incorrect_source_profile).unwrap();
+        writeln!(config_file, "region={}", &get_test_region()).unwrap();
+        (correct, incorrect)
+    };
+
+    config_file.flush().unwrap();
+
+    // Set up the environment variables to use this new config file. This is only OK to do because
+    // this test is run in a forked process, so won't affect any other concurrently running tests.
+    std::env::set_var("AWS_CONFIG_FILE", config_file.path().as_os_str());
+
+    // With correct profile, things should be fine
+    let config = S3ClientConfig::new()
+        .auth_config(S3ClientAuthConfig::Profile(correct_profile.to_owned()))
+        .endpoint_config(EndpointConfig::new(&get_test_region()));
+    let client = S3CrtClient::new(config).unwrap();
+    let _result = client
+        .list_objects(&bucket, None, "/", 10, &format!("{prefix}foo/"))
+        .await
+        .expect("list_objects should succeed");
+
+    // With incorrect profile, requests should fail with a client error
+    let config = S3ClientConfig::new()
+        .auth_config(S3ClientAuthConfig::Profile(incorrect_profile.to_owned()))
+        .endpoint_config(EndpointConfig::new(&get_test_region()));
+    let client = S3CrtClient::new(config).unwrap();
+    let err = client
+        .list_objects(&bucket, None, "/", 10, &format!("{prefix}/"))
+        .await
+        .expect_err("should fail in different prefix");
```

Comment:
nit: update the expect message here

```suggestion
        .expect_err("should fail when using invalid credentials");
```